### DISCUSSION TITLE
threading model

### DISCUSSION THREAD

**Discussion by Infixz** [apernet/hysteria#788] [Q&A]
is that hysteria using 3 threads as default? i noticed it will spawn more thread when one thread blocked due to network, but will it kill the dead thread or kill the spawn'ed thread after previous one resumed?

**Comment by tobyxdd** [comment]
It spawns 2 goroutines per client connection, but goroutines are not mapped 1-1 to system threads. You can read more about how goroutines are scheduled here: https://betterprogramming.pub/deep-dive-into-concurrency-of-go-93002344d37b
